test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -133,3 +123,47 @@
         (testing (format "%s requests are refused" (name v)))
         (is (retirement-response? v (ping v)))))))
 
+(deftest whitelist-middleware
+  (testing "should log on reject"
+    (let [wl (fs/temp-file "whitelist-log-reject")]
+      (.deleteOnExit wl)
+      (spit wl "foobar")
+      (let [f (build-whitelist-authorizer (fs/absolute-path wl))]
+        (is (= :authorized (f {:ssl-client-cn "foobar"})))
+        (with-log-output logz
+          (is (string? (f {:ssl-client-cn "badguy"})))
+          (is (= 1 (count (logs-matching #"^badguy rejected by certificate whitelist " @logz)))))))))
+
+(defn make-https-request-with-whitelisted-host [whitelisted-host]
+  (let [whitelist-file (fs/temp-file "whitelist-log-reject")
+        cert-config {:ssl-cert "test-resources/localhost.pem"
+                     :ssl-key "test-resources/localhost.key"
+                     :ssl-ca-cert "test-resources/ca.pem"}
+        config (-> (svc-utils/create-config)
+                   (assoc :jetty (merge cert-config
+                                        {:ssl-port 0
+                                         :ssl-host "0.0.0.0"
+                                         :ssl-protocols "TLSv1,TLSv1.1"}))
+                   (assoc-in [:puppetdb :certificate-whitelist] (str whitelist-file)))]
+    (.deleteOnExit whitelist-file)
```

Comment:
We have a utils function for creating a file that gets deleted on jvm exit I believe.